### DISCUSSION TITLE
Is it possible to tunnel all device?

### DISCUSSION THREAD

**Discussion by parsanoori** [shadowsocks/shadowsocks-org#212] [Q&A]
Can we use shadowsocks client to tunnel all device traffic to shadowsocks server? How?

**Comment by RnJ4** [comment]
If you are using windows,<del>I would suggest using TUN mode from [Clash for windows](https://github.com/Fndroid/clash_for_windows_pkg/releases).</del>
Outline might be a better option if you are living Outside of China

**Comment by robot-dot-win** [comment]
> Can we use shadowsocks client to tunnel all device traffic to shadowsocks server? How?

You can use Clash/V2Ray/Xray. But it's a little bit complex.

If you are running shadowsocks on a Linux box, you can easily set up a transparent proxy for all devices via iptables/nftables rules and the `local-redir` feature of [shadowsocks-rust](https://github.com/shadowsocks/shadowsocks-rust).
